### PR TITLE
KAFKA-15614: [Minor] Renaming getter and correcting code comments

### PR DESCRIPTION
**Pull request by apoorvmittal10**
This PR is follow-up for minor remaining comments from https://github.com/apache/kafka/pull/14575

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/server/telemetry/ClientTelemetryPayload.java [hunk 1]
```diff
@@ -22,31 +22,37 @@
...
 
     /**
+     * Method returns the client's instance id.
+     *
      * @return Client's instance id.
      */
     Uuid clientInstanceId();
 
     /**
      * Indicates whether client is terminating, e.g., the last metrics push from this client instance.
```
Comment (AndrewJSchofield) [inline clients/src/main/java/org/apache/kafka/server/telemetry/ClientTelemetryPayload.java]: "Indicates whether the client is terminating and thus making its last metrics push."

File: clients/src/main/java/org/apache/kafka/server/telemetry/ClientTelemetryPayload.java [hunk 2]
```diff
@@ -22,31 +22,37 @@
...
-     *To avoid the receiving broker’s metrics rate-limiter discarding this out-of-profile push, the
-     * PushTelemetryRequest.Terminating field must be set to true. A broker must only allow one such
-     * unthrottled metrics push for each combination of client instance ID and SubscriptionId.
      *
      * @return {@code true} if client is terminating, else false
      */
     boolean isTerminating();
 
     /**
+     * Method returns the content-typ format of the metrics data which is being sent by the client.
```
Comment (AndrewJSchofield) [inline clients/src/main/java/org/apache/kafka/server/telemetry/ClientTelemetryPayload.java]: Typo. Should be `content-type`.

File: clients/src/main/java/org/apache/kafka/server/telemetry/ClientTelemetryPayload.java (hunk 1 above)
Comment (apoorvmittal10) [reply clients/src/main/java/org/apache/kafka/server/telemetry/ClientTelemetryPayload.java]: Done.

File: clients/src/main/java/org/apache/kafka/server/telemetry/ClientTelemetryPayload.java (hunk 2 above)
Comment (apoorvmittal10) [reply clients/src/main/java/org/apache/kafka/server/telemetry/ClientTelemetryPayload.java]: My bad.

**Review by AndrewJSchofield — APPROVED** [review]
lgtm

**Comment by mjsax** [comment]
Thanks for the follow up. Merged to `trunk`.

**Pull request merged**